moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -534,30 +544,47 @@ KinematicState Servo::getCurrentRobotState()
   return current_state;
 }
 
-std::pair<bool, KinematicState> Servo::smoothHalt(KinematicState halt_state)
+std::pair<bool, KinematicState> Servo::smoothHalt()
 {
   bool stopped = false;
   const auto current_state = getCurrentRobotState();
 
-  const size_t num_joints = current_state.joint_names.size();
-  for (size_t i = 0; i < num_joints; i++)
+  if (last_commanded_state_)
   {
-    const double vel = (halt_state.positions[i] - current_state.positions[i]) / servo_params_.publish_period;
-    halt_state.velocities[i] = (vel > STOPPED_VELOCITY_EPS) ? vel : 0.0;
-    halt_state.accelerations[i] =
-        (halt_state.velocities[i] - current_state.velocities[i]) / servo_params_.publish_period;
-  }
+    auto& halt_state = last_commanded_state_.value();
+    const size_t num_joints = current_state.joint_names.size();
+    for (size_t i = 0; i < num_joints; i++)
+    {
+      const double vel = (halt_state.positions[i] - current_state.positions[i]) / servo_params_.publish_period;
+      halt_state.velocities[i] = (vel > STOPPED_VELOCITY_EPS) ? vel : 0.0;
+      halt_state.accelerations[i] =
+          (halt_state.velocities[i] - current_state.velocities[i]) / servo_params_.publish_period;
+    }
 
-  // If all velocities are zero, robot has decelerated to a stop.
-  stopped = (std::accumulate(halt_state.velocities.begin(), halt_state.velocities.end(), 0.0) == 0.0);
+    // If all velocities are zero, robot has decelerated to a stop.
+    stopped = (std::accumulate(halt_state.velocities.begin(), halt_state.velocities.end(), 0.0) == 0.0);
```

Comment:
In practice, i think there will always be some noise so this will prob fail

Reply:
Ah yes, the specific noise level would also depend on hardware I guess.
Would checking `<=` some threshold like `0.00001` be enough to account for the noise ?

Reply:
Agreed. There already is a variable for this, so I'll update it in my bug fix PR.